### PR TITLE
[BugFix] Fix aclgraph accu problem in A2.

### PR DESCRIPTION
**Pull request by whx-sjtu**
This PR fixes accuracy problem of aclgraph on A2. The problem is introduced by PR #2980, which makes the `all_reduce` of shared_experts exposed to torch dynamo. This PR moves all the codes into forward_impl to shiled from torch dynamo.

- vLLM version: v0.10.2
- vLLM main: https://github.com/vllm-project/vllm/commit/17b4c6685ce62d5652654784d6771a3d38e4273e

### CODE REVIEW

**Comment by shen-shanshan** [comment]
LGTM.

**Comment by yiz-liu** [comment]
LGTM

File: vllm_ascend/ops/common_fused_moe.py [hunk 1]
```diff
@@ -347,27 +353,32 @@ def forward(
...
-        fused_output = AscendFusedMoE.forward_impl(
-            self,
-            hidden_states=hidden_states,
-            router_logits=router_logits,
-        )
-        return shared_output, fused_output
 
+direct_register_custom_op(op_name="maybe_all_reduce_tensor_model_parallel",
+                          op_func=_maybe_all_reduce_tensor_model_parallel_impl,
+                          fake_impl=lambda x, label: x,
```
Comment (realliujiaxu) [inline vllm_ascend/ops/common_fused_moe.py]: remove argument `label`, as there is no such an arg in `_maybe_all_reduce_tensor_model_parallel_impl`

File: vllm_ascend/ops/common_fused_moe.py [hunk 2]
```diff
@@ -347,27 +353,32 @@ def forward(
...
             router_logits=router_logits,
         )
         # Make sure the default stream waits for the shared experts stream to finish.
         if self.multistream_overlap_shared_expert:
             torch.npu.current_stream().wait_stream(self.shared_expert_stream)
-        return shared_out, fused_out
+        return shared_out, fused_output
+
+
+def _maybe_all_reduce_tensor_model_parallel_impl(
```
Comment (realliujiaxu) [inline vllm_ascend/ops/common_fused_moe.py]: move this to register_custom_ops.py

**Review by realliujiaxu — APPROVED** [review]

**Review by zzzzwwjj — APPROVED** [review]

**Comment by Yikun** [comment]
1. This PR doesn't include any e2e tests, please describe how you tested it.
2. What are your plans for adding e2e test to prevent this issue from breaking again? Please add the test plan as issue.

**Comment by wangxiyuan** [comment]
@Yikun talked with @whx-sjtu the case is that A2 + aclgraph  + EP + max_num_tokens>512  our CI doesn't cover this case. Let's add one later.
